src/scriptdev2/scripts/eastern_kingdoms/zulgurub/boss_jeklik.cpp

```diff
@@ -215,24 +238,51 @@ struct boss_jeklikAI : public ScriptedAI
             if (m_uiSwoopTimer < uiDiff)
             {
                 if (DoCastSpellIfCan(m_creature->getVictim(), SPELL_SWOOP) == CAST_OK)
+                {
+                    if (m_creature->getThreatManager().getThreat(m_creature->getVictim()))
+                        m_creature->getThreatManager().modifyThreatPercent(m_creature->getVictim(), -10);
+
                     m_uiSwoopTimer = urand(4000, 9000);
+                }
             }
             else
                 m_uiSwoopTimer -= uiDiff;
 
             if (m_uiSonicBurstTimer < uiDiff)
             {
                 if (DoCastSpellIfCan(m_creature, SPELL_SONIC_BURST) == CAST_OK)
+                {
                     m_uiSonicBurstTimer = urand(8000, 13000);
+                }
             }
             else
                 m_uiSonicBurstTimer -= uiDiff;
 
+            if (m_uiPsyhicScreamTimer < uiDiff)
+            {
+                if (DoCastSpellIfCan(m_creature, SPELL_PSYHIC_SCREAM) == CAST_OK)
+                {
+                    if (m_creature->getThreatManager().getThreat(m_creature->getVictim()))
+                        m_creature->getThreatManager().modifyThreatPercent(m_creature->getVictim(), -80);
+                    m_uiPsyhicScreamTimer = urand(18000, 26000);
+                }
+            }
+            else
+                m_uiPsyhicScreamTimer -= uiDiff;
+
             if (m_uiSpawnBatsTimer < uiDiff)
             {
-                if (DoCastSpellIfCan(m_creature, SPELL_SUMMON_FRENZIED_BATS) == CAST_OK)
+                DoScriptText(SAY_SHRIEK, m_creature);
+                if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
                 {
-                    DoScriptText(SAY_SHRIEK, m_creature);
+                    for (uint8 i = 0; i < 6; ++i)
+                    {
+                        //DoCastSpellIfCan(pTarget, SPELL_SUMMON_FRENZIED_BATS);
+                        if (Creature* bat = m_creature->SummonCreature(NPC_FRENZIED_BAT, -12291.6220f, -1380.2640f, 144.8304f, 5.483f, TEMPSUMMON_DEAD_DESPAWN, 0)) {
+                            bat->SetLevitate(true);
```

Comment:
Main problem here: why not use the spell, instead of the summon function?
We are using spells for a reason, and when we don't it's usually because it's really too complicated to fix the spell.